### PR TITLE
New semi-dynamic NPC

### PR DESCRIPTION
**Pull request by SirPiecemaker**
```SUMMARY: Content "Adds a new, somewhat dynamic NPC"```

#### Purpose of change

I felt many of the NPC's felt the same in their interactions, so I tried to make an NPC that behaves differently in their interactions with the player, namely conversation and things like scamming them for gear. 

#### Describe the solution

Added an NPC and painstakingly made most of the normal interactions by hand to be unique. While I do have ideas for more for the character, I was afraid people will dislike it and I would waste my time on additional content like it's own, minor faction. 

That being said, it got a little out of hand and I ended up making it more interaction-reliant than I intended. It's certainly not Foodperson level, but it might be somewhere in between.

#### Additional context

Apologies for the probably numerous typos as my keyboard is broken and often fails.

I also want to add that the NPC does not have any special, mystical powers as to not break the setting - it is merely a belief it holds.

### CODE REVIEW

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+      "id": "NC_BONE_SEER",
+      "name": "Bone Seer",
+      "job_description": "I wish to sing the song of the Bones",
+      "traits": [
+        { "trait": "OUTDOORSMAN" },
+        { "trait": "HEAVYSLEEPER" },
+        { "trait": "GOODCARDIO" },
+        { "trait": "LIGHTSTEP" },
+        { "trait": "ROBUST" },
+		{ "trait": "seer_mark" },
```
Comment (mlangsdorf) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
	 	{ "trait": "seer_mark" },
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      "id": "SEER_Brigitte_LaCroix",
+      "//": "A mysterious member of a doomsday cult.",
+      "name_unique": "Brigitte LaCroix",
+      "gender": "female",
+      "name_suffix": "Bone Seer",
+      "class": "NC_BONE_SEER",
+      "attitude": 7,
+      "mission": 3,
+      "chat": "TALK_BONE_SEER",
+      "faction": "wasteland_scavengers",
```
Comment (mlangsdorf) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]: We're discouraging the use of "wasteland_scavengers" as a catch-all faction for NPCs.  She could either have no faction, or her own faction with her being the only member.  The first option is easier.

Comment (SirPiecemaker) [reply data/json/npcs/NPC_Brigitte_LaCroix.json]: When I left it open it gave me ann error

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+        [ "broth_bone", 60 ],
+        [ "water_clean", 90 ]
+      ]
+    },
+	{
+    "type": "mutation",
+    "id": "seer_mark",
+    "name": "Mark of the Seer",
+    "points": 1,
+    "description": "A strange occult symbol carved into your hand by Brigitte LaCroix",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
    "description": "A strange occult symbol carved into your hand by Brigitte LaCroix.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+        { "text": "I have to get going. Take care, Seer.", "topic": "TALK_DONE" }
+      ]
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_SONG",
+      "dynamic_line": "Listen carefully. The bones... they sing. Can you hear it? The song they weave? The stories they hold?",
+      "responses": [
+        { "text": "What? Singing bones? What are you talking about?", "topic": "TALK_BONE_SEER_SONG2" },
+        { "text": "Uh...yeah, sure. I think I'll just be on my way.", "topic": "TALK_DONE" }
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
        { "text": "Uh... yeah, sure. I think I'll just be on my way.", "topic": "TALK_DONE" }
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+      ]
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_SONG2",
+      "dynamic_line": "When it all happened, the cataclysm, something... changed. You can see it in all creatures, but most of all their bones. They break, morph, rise again, in an infinite cycle. Living dead walk. Monsters rip and tear each other apart. You can see the resonance, the quiet hum of raw strength, and only by taking the bones does the cycle end - their story, their song, their strength, become yours to use.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "dynamic_line": "When it all happened, <the_cataclysm>, something... changed. You can see it in all creatures, but most of all their bones. They break, morph, rise again, in an infinite cycle. Living dead walk. Monsters rip and tear each other apart. You can see the resonance, the quiet hum of raw strength, and only by taking the bones does the cycle end - their story, their song, their strength, become yours to use.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+      "responses": [
+        { "text": "So what do you actually do with the bones?", "topic": "TALK_BONE_SEER_BONES" },
+        { "text": "I'm still not quite sure what you mean about songs or strength.", "topic": "TALK_BONE_SEER_SONG3" },
+        { "text": "Does anyone else, uh, believe this?", "topic": "TALK_BONE_SEER_OTHERS" }
+      ]
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_SONG3",
+      "dynamic_line": "Only when you crush the bones of a body does it cease to rise. Only if you examine the bones can you see what was. Thus is the story. Whatever causes this change is alive, moving within us all, an inevitable part of this new world. It holds the power of change. When we hold the bones, we hold the power. Thus the strength. Together...they form a beautiful song.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "dynamic_line": "Only when you crush the bones of a body does it cease to rise. Only if you examine the bones can you see what was. Thus is the story. Whatever causes this change is alive, moving within us all, an inevitable part of this new world. It holds the power of change. When we hold the bones, we hold the power. Thus the strength. Together... they form a beautiful song.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+      "dynamic_line": "The song can be weaved in many forms. Carved bone charms, weapons and armor all hold immense power, and when the time comes, me and my kindred shall gather a great amount of song and sing it to restore this world. Restore it, or end it. Makes no difference.",
+      "responses": [ { "text": "End the world? What?", "topic": "TALK_BONE_SEER_END" } ]
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_END",
+      "dynamic_line": "We believe that enough power in one song could revert the Cataclysm - or accelerate it to a time beyond all, ending it all the same. But with the world looking as is, both options are preferable.",
+      "responses": [
+        {
+          "text": "Sounds somewhat...doomsday-ish, but I suppose it's a belief like any other. Still, at least you have a goal to help the world, one way or another.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
          "text": "Sounds somewhat... doomsday-ish, but I suppose it's a belief like any other. Still, at least you have a goal to help the world, one way or another.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+          "topic": "TALK_BONE_SEER_ACOLYTE"
+        },
+        { "text": "You're nuts.", "topic": "TALK_BONE_SEER_INSULT" }
+      ]
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_ACOLYTE",
+      "dynamic_line": "Your mind is open. More than most. Perhaps one day, you too will feel the power of the song and become Kindred. For now, Acolyte, listen, listen and feel the song.",
+      "responses": [ { "text": "I...thank you.", "topic": "TALK_BONE_SEER" } ]
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "responses": [ { "text": "I... thank you.", "topic": "TALK_BONE_SEER" } ]
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+    },
+    {
+      "type": "talk_topic",
+      "id": "TALK_BONE_SEER_INSULT",
+      "dynamic_line": "Your skepticism does not surprise me. Perhaps one day, you too will hear the inevitability of the song, feel it's power. But until then, you will remain an Acolyte, path to the Kindred closed.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "dynamic_line": "Your skepticism does not surprise me. Perhaps one day, you too will hear the inevitability of the song, feel its power. But until then, you will remain an Acolyte, path to the Kindred closed.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+      "responses": [ { "text": "Yeah, alright.", "topic": "TALK_BONE_SEER" } ]
+    },
+    {
+      "id": "MISSION_SEER_GATHER_BONE",
+      "type": "mission_definition",
+      "name": "Gather bone for the Brigitte. About 8 should do it.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "name": "Gather bones for the Brigitte. About 8 should do it.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+      "difficulty": 1,
+      "value": 10000,
+      "goal": "MGOAL_FIND_ITEM",
+      "item": "bone",
+      "count": 8,
+      "origins": [ "ORIGIN_SECONDARY" ],
+      "end": { "effect":[ { "u_add_var": "NC_SEER_MISSION_1", "type": "general", "context": "mission", "value": "yes", "u_add_trait": "seer_mark" } ] },
+      "dialogue": {
+        "describe": "There is always work to be done, song to be woven.",
+        "offer": "If you wish to be set on the path to enlightenment, first you must learn to listen and hear the song. Go out, butcher a creature and feel the power between your fingertips. Then bring me the bones and I shall carve them for you. ",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
        "offer": "If you wish to be set on the path to enlightenment, first you must learn to listen and hear the song. Go out, butcher a creature and feel the power between your fingertips. Then bring me the bones and I shall carve them for you.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+        "failure": "Then you shall try again, until you hear."
+      }
+    },
+	{
+    "id": "TALK_MISSION_LIST_SEER",
+    "type": "talk_topic",
+    "category": "CATEGORY_MISSION_SEER",
+    "dynamic_line": {
+      "has_no_available_mission": {
+        "has_no_assigned_mission": "The song is...quiet for now. Perhaps with time, more notes will be etched in the bones of this world.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
        "has_no_assigned_mission": "The song is... quiet for now. Perhaps with time, more notes will be etched in the bones of this world.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+	{
+	 "id": "TALK_SUGGEST_FOLLOW_SEER",
+     "type": "talk_topic",
+	 "dynamic_line": {
+	 "u_has_trait": "seer_mark",
+     "type": "dialogue",
+     "context": "mark_of_the_seer",
+     "value": "yes",
+     "no": "Only those who bear my mark will prove themselves worthy of my skills.",
+	 "yes": "You bear my mark, meaning I believe you have potentional to learn to truly listen to the Song. Yes, I will lend my skills to you, for now."
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
	 "yes": "You bear my mark, meaning I believe you have potential to learn to truly listen to the Song. Yes, I will lend my skills to you, for now."
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+	 "dynamic_line": "I understand your reluctancy. Feel free to return when you see the way.",
+	 "responses": [
+	 { "text": "Maybe some other time. Changing the topic...", "topic": "TALK_NONE" },
+	 { "text": "Alright, but I have to go now.", "topic": "TALK_DONE" }
+	 ]
+	 },
+	 {
+	  "type": "talk_topic",
+      "id": "TALK_BONE_SEER_CYCLE",
+      "dynamic_line": "It's not just walking horrors and monsters that have changed with the Cataclysm. It started a...cycled, of sorts. Everything repeats. We can only see it in others, but it happens to us, even you and I. How many times have you fallen? Your flesh rent from your body, devoured. Or perhaps it was the quiet whimper of death to exposure. But your bones rose again. Different flesh, different name, sometimes even different knowledge, but the bones, the same. We are all trapped in the same cycle. We just keep forgetting. That's why we need to amass the Song. That's why it has to end, even if it means the destruction, not restoration.",
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
      "dynamic_line": "It's not just walking horrors and monsters that have changed with the Cataclysm. It started a... cycle, of sorts. Everything repeats. We can only see it in others, but it happens to us, even you and I. How many times have you fallen? Your flesh rent from your body, devoured. Or perhaps it was the quiet whimper of death to exposure. But your bones rose again. Different flesh, different name, sometimes even different knowledge, but the bones, the same. We are all trapped in the same cycle. We just keep forgetting. That's why we need to amass the Song. That's why it has to end, even if it means the destruction, not restoration.",
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+      "responses": [
+        { "text": "That is...that's really one hell of a belief. Well, if it helps you deadl with the world, who am I to argue.", "topic": "TALK_NONE" },
```
Comment (Night-Pryanik) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
        { "text": "That is... that's really one hell of a belief. Well, if it helps you deal with the world, who am I to argue.", "topic": "TALK_NONE" },
```

File: data/json/npcs/NPC_Brigitte_LaCroix.json (hunk 5 above)
Comment (SirPiecemaker) [reply data/json/npcs/NPC_Brigitte_LaCroix.json]: I greatly appreciate all the checking you do. I try to check it mmyself, but sometimes when you write it you sort of fix it unconciously in your brain, plus the keyboard is busted.

Inn this case Cataclysm is actually a stylistic choice. The NPC has a particular way of talking and some of the snippets would not fit her character.

Comment (Night-Pryanik) [reply data/json/npcs/NPC_Brigitte_LaCroix.json]: Ok, no problems.

File: data/json/npcs/NPC_Brigitte_LaCroix.json (hunk 15 above)
Comment (SirPiecemaker) [reply data/json/npcs/NPC_Brigitte_LaCroix.json]: Thanks for the fixes!

**Comment by SirPiecemaker** [comment]
Is there anything I should add/change with this PR for merge?

**Comment by kevingranade** [comment]
Ok, so they're a crazy person that's obsessed with bones, that's weird but ok.
You gave them a bunch of deprecated bone gear, which is not.
Bone armor of all kinds is obsoleted and on the way out, see https://github.com/CleverRaven/Cataclysm-DDA/pull/28064

**Comment by SirPiecemaker** [comment]
> Ok, so they're a crazy person that's obsessed with bones, that's weird but ok.
> You gave them a bunch of deprecated bone gear, which is not.
> Bone armor of all kinds is obsoleted and on the way out, see #28064

That was the point - everyone else in the apocalypse seems to be far too logical. This is at least a bit more memorable than just random survivor. As for the armor, don't know what I am supposed to do. Just give her different armor?

**Comment by SirPiecemaker** [comment]
Replaced the bone armor with more generic gear. Please let me know if anything else should be tweaked. Much obliged.

File: data/json/npcs/NPC_Brigitte_LaCroix.json [hunk 16]
```diff
@@ -0,0 +1,372 @@
...
+    "type": "item_group",
+    "id": "BONE_SEER_Brigitte_wield",
+    "subtype": "collection",
+    "entries": [ { "item": "2h_flail_steel" } ]
+  },
+  {
+    "type": "item_group",
+    "id": "NC_BONE_SEER_misc",
+    "items": [
+      [ "bone_plate", 10 ],
```
Comment (kevingranade) [inline data/json/npcs/NPC_Brigitte_LaCroix.json]:
```suggestion
```

File: data/json/mapgen/cemetery_4square.json [hunk 17]
```diff
@@ -413,7 +413,8 @@
         { "item": "church", "x": [ 14, 18 ], "y": [ 5, 14 ], "chance": 65 },
         { "item": "church", "x": [ 11, 12 ], "y": [ 5, 6 ], "chance": 65 },
         { "item": "jackets", "x": [ 12 ], "y": [ 14 ], "chance": 50 }
-      ]
+      ],
+      "place_npcs": [ { "chance": 7, "class": "SEER_Brigitte_LaCroix", "x": 18, "y": 6 } ]
```
Comment (jbytheway) [inline data/json/mapgen/cemetery_4square.json]: FYI, `chance` is not a supported field in `place_npcs` (although perhaps it should be), and so this NPC will always spawn here.

**Comment by Brytenwalda** [comment]
Does this NPC actually spawn? I’m asking as a modder - how and what conditions are set for this NPC to spawn?

File: data/json/mapgen/cemetery_4square.json (hunk 17 above)
Comment (Brytenwalda) [reply data/json/mapgen/cemetery_4square.json]: How else should it be done?

Comment (jbytheway) [reply data/json/mapgen/cemetery_4square.json]: I believe the only way to get a non-certain probability of NPC spawning right now is to use nested mapgen.  However, I think adding support for `chance` is probably the better solution in the long term.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/is-npc-spawning-a-yes-no-thing/24443/4

**Pull request merged**